.github/workflows/release-step-3.yml

```diff
@@ -213,17 +213,28 @@ jobs:
     runs-on: ubuntu-latest
     needs:
       - post-release
-    permissions:
-      contents: write
-    env:
-      GITHUB_TOKEN: ${{ secrets.RELEASE_GITHUB_TOKEN }}
     steps:
+      - name: Get token
+        id: get_token
+        uses: tibdex/github-app-token@3beb63f4bd073e61482598c45c71c1019b59b73a # v2.1.0
+        with:
+          app_id: ${{ secrets.OBS_AUTOMATION_APP_ID }}
+          private_key: ${{ secrets.OBS_AUTOMATION_APP_PEM }}
+          permissions: >-
+            {
+              "contents": "write"
```

Comment:
IIUC, this step is only for updating a GitHub branch, hence the permission has been changed

Reply:
wrong assumption:
- https://github.com/elastic/apm-agent-java/blob/9a12b98ee959abb4a596c823325d9c78d827724b/.ci/release/update-major-branch.sh#L70-L75
It creates a PR too

Reply:
```suggestion
              "contents": "write",
              "pull_requests": "write"
```